rasa/nlu/utils/hugging_face/hf_transformers.py

```diff
@@ -684,9 +688,9 @@ def _get_docs_for_batch(
 
         return batch_docs
 
-    def train_chunk(
+    def train(
         self,
-        training_data_chunk: TrainingDataChunk,
+        training_data: TrainingDataFull,
```

Comment:
@dakshvar22 should this be `train_chunk`?

Reply:
Hmm, I think because components like `HFTransformersNLP`, and `SpacyNLP` go from creating tokens to features inside one component, it doesn't need a separate `train_chunk` method explicitly because we are passing the complete training data [here](https://github.com/RasaHQ/rasa/pull/7551/files#diff-178f448a45faaee012886c9ac7564a55639e9ff7383e2c68ae4c1210020276daR285). But then those components should be doing the chunking themselves and featurizing chunks? These components are really an exception for the logic in the component class and break the abstraction between `train` and `train_chunk` I think.

Reply:
but the problem is that they will create dense features for the whole training data, which will blow up in memory. I think the problem is that they are applied before tokenizers

Reply:
Yes, that's what I meant by these components being an exception for the logic in `component.py`

Reply:
let's add them to the to do list, and tackle it in the separate PR